agent-inject/agent/annotations.go

```diff
@@ -123,12 +123,15 @@ const (
 	// AnnotationVaultAuthPath specifies the mount path to be used for the Kubernetes auto-auth
 	// method.
 	AnnotationVaultAuthPath = "vault.hashicorp.com/auth-path"
+
+	// AnnotationVaultAuthMethod specifies the auto-auth method to use
+	AnnotationVaultAuthMethod = "vault.hashicorp.com/auth-method"
```

Comment:
would it be better to have auth config instead, cause there might be a need for other additional params in future? What do you think?